frontend/src/components/applyForm/utils.tsx

```diff
@@ -431,29 +438,49 @@ export const buildField = ({
   });
 };
 
-const formatFieldWarnings = (
+const getNestedWarningsForField = (
+  fieldName: string,
+  warnings: FormValidationWarning[],
+): FormValidationWarning[] => {
+  return warnings.filter(({ field, type }) => {
+    return type === "required" && fieldName.includes(field);
+  });
+};
+
+const getWarningsForField = (
+  fieldName: string,
+  isRequired: boolean,
+  warnings: FormValidationWarning[],
+): FormValidationWarning[] => {
+  const directWarnings = warnings?.filter(
+    (warning) => warning.field.indexOf(fieldName) !== -1,
+  );
+  const nestedRequiredWarnings = isRequired
+    ? getNestedWarningsForField(fieldName, warnings)
+    : [];
+  return [...directWarnings, ...nestedRequiredWarnings];
+};
+
+export const formatFieldWarnings = (
   warnings: FormValidationWarning[] | null,
   name: string,
-  type: string,
+  fieldType: string,
+  required: boolean,
 ): string[] => {
   if (!warnings || warnings.length < 1) {
     return [];
   }
-  if (type === "array") {
-    const data = warnings.reduce(
+  if (fieldType === "array") {
+    const warningMap = warnings.reduce(
       (acc, item) => {
-        const field = item.field.replace(/^\$\./, "");
```

Comment:
since the error fields are already getting formatted this way here, this is unnecessary. There are still bugs in this array code, though, which we'll handle in a separate ticket tbd